### ISSUE TITLE
sort out qt4 v qt5

### ISSUE THREAD

**Issue by tacaswell** [conda-forge/matplotlib-feedstock#59]
The upstream issue of https://github.com/ContinuumIO/anaconda-issues/issues/1068 is going to hit conda-forge soonish.

The rough issue is that as part of the mpl build the rctemplate is patched to be hard-coded to qt4Agg as the backend.  Continuum seems to be shipping qt5 now which is causing havoc.

attn @pelson

**Comment by ocefpaf** [comment]
Some feedstock are already failing b/c they are mixing `pyqt 5.*` (compiled against `qt 5.*` from `defaults`). That is happening on Windows right now thanks to @ccordoba12 who built `Qt 4` and `pyqt 4` for Linux and `pyqt 4` for `OS X` for conda-forge.

- https://github.com/conda-forge/mplleaflet-feedstock/pull/3
- https://github.com/conda-forge/modflow2netcdf-feedstock/pull/3
- https://github.com/conda-forge/windrose-feedstock/pull/3
- https://github.com/conda-forge/flopy-feedstock/pull/6

I believe that pining `pyqt` to `4.11.*` would be an OK workaround until we have `Qt` and `pyqt`  `4` and `5` built for all platforms in `conda-forge`. See #72

**Comment by megies** [comment]
+1 just set up a fresh environment and ended up with qt/pyt 5 and mpl 1.5.3 which resulted in the following import error:
```
    import matplotlib.pyplot as plt
  File "/home/megies/anaconda2/envs/default/lib/python2.7/site-packages/matplotlib/pyplot.py", line 114, in <module>
    _backend_mod, new_figure_manager, draw_if_interactive, _show = pylab_setup()
  File "/home/megies/anaconda2/envs/default/lib/python2.7/site-packages/matplotlib/backends/__init__.py", line 32, in pylab_setup
    globals(),locals(),[backend_name],0)
  File "/home/megies/anaconda2/envs/default/lib/python2.7/site-packages/matplotlib/backends/backend_qt4agg.py", line 18, in <module>
    from .backend_qt5agg import FigureCanvasQTAggBase as _FigureCanvasQTAggBase
  File "/home/megies/anaconda2/envs/default/lib/python2.7/site-packages/matplotlib/backends/backend_qt5agg.py", line 16, in <module>
    from .backend_qt5 import QtCore
  File "/home/megies/anaconda2/envs/default/lib/python2.7/site-packages/matplotlib/backends/backend_qt5.py", line 31, in <module>
    from .qt_compat import QtCore, QtGui, QtWidgets, _getSaveFileName, __version__
  File "/home/megies/anaconda2/envs/default/lib/python2.7/site-packages/matplotlib/backends/qt_compat.py", line 137, in <module>
    from PyQt4 import QtCore, QtGui
ImportError: No module named PyQt4
```

I can just pin qt/pyqt to 4.* and it works.. but probably the matplotlib conda package should depend on qt/pyqt <5, so that less experienced users don't end up with a broken environment..?

**Comment by tacaswell** [comment]
It pins to 4.11.* : https://github.com/conda-forge/matplotlib-feedstock/blob/master/recipe/meta.yaml#L59 as of 2.0.  There might be some old builds floating around that are not sufficiently pinned?

**Comment by megies** [comment]
I just tried again and indeed, I can confirm this is fixed now by a new mpl 1.5 build: :+1: 
```
$ conda install 'matplotlib=1.5' 'pyqt=5' qt
Fetching package metadata ...........
Solving package specifications: ....


UnsatisfiableError: The following specifications were found to be in conflict: 
  - matplotlib 1.5*
  - qt 4.*
Use "conda info <package>" to see the dependencies for each package.
```

**Comment by ocefpaf** [comment]
I guess we can close this. We have successfully pinned all of `conda-forge` to `qt/pyqt 4`.

Now that we do have `Qt 5` on both Windows and Linux and I am slowly building the packages with them.

Please ping me if anyone finds any issues during this move.
